### PR TITLE
Attempt to fix layer ordering

### PR DESCRIPTION
**Pull request by johnarban**
## Depth ordering 
This builds off of @Carifio24's re-ordering of the layer draw order. Added `SpreadSheetLayer.draw` to `wwt-hacks`
1)  set `this.pointList.depthBuffered = true;`.
2) set `shallowLayerTest` to just be `(layer) => true`. (We always want the layer to load after the planets

At this point the points all are correctly ordered, _but_ they clip each other. 

3) wrap the `pointList` draw commands with `renderContext.gl.depthMask(false)`/`renderContext.gl.depthMask(true)`. _I just saw this command online and tried it. Had no clue why it worked_

I think the principle (based on links in the code comments) is: that `gl.enable(WEBGL.DEPTH_TEST`) means that we know the depth of the Planets, and so if any point falls behind the planet, it gets discarded. (drawing is done point by point, so it knows where each one is. So we don't need to chunk the data anymore). Toggling `gl.depthMask` makes sure the points don't write their depth to depth buffer so they never clip each other. 

The solution seems to work here, I don't know if it s a general solution, as I don't know if it will break. The web engine never uses `gl.depthMask`, but it's been supported since 2015. 

## What it breaks
This does break something though. The planet points (drawn with `TimeSeriesPointSpriteShader`, same as the `SpreadSheetLayer`, both use `PointList`) now ignore their transparency. 
image

I could not figure out a combination of the two commands to fix this.

### CODE REVIEW

File: src/WwtPlayground.vue [hunk 1]
```diff
@@ -559,36 +481,22 @@ onMounted(() => {
...
-      const centerRow = rows[center];
-      const x = Number(centerRow[layer.get_xAxisColumn()]);
-      const y = Number(centerRow[layer.get_yAxisColumn()]);
-      const z = Number(centerRow[layer.get_zAxisColumn()]);
-      // @ts-expect-error this does exist
-      const depth = WWTControl.singleton.getDepth(x, y, z);
-      // @ts-expect-error this does exist
-      const moonDepth = WWTControl.singleton.getDepth(0, 0, 0);
-      return depth <= (moonDepth + 0.0000116 / 6); // magic number minor improvement to front side depth test
+    WWTControl.singleton.shallowLayerTest = function(_layer) {
```
Comment (Carifio24) [inline src/WwtPlayground.vue]: I think if we don't need the shallow layer test we should either set it to `null` (or remove that logic in the render loop) so that we aren't running the layer filter twice.

**Review by Carifio24 — CHANGES_REQUESTED** [review]

**Comment by Carifio24** [comment]
I also don't quite understand the part about wrapping the point layer draw calls in `gl.depthMask` updates, but this seems to work correctly here.

File: src/WwtPlayground.vue (hunk 1 above)
Comment (johnarban) [reply src/WwtPlayground.vue]: Alright, yeah. I will put a commit to do that

**Comment by johnarban** [comment]
> I also don't quite understand the part about wrapping the point layer draw calls in `gl.depthMask` updates, but this seems to work correctly here.

yeah, i am not really sure either, i just got lucky in google searches 🍀 .  I think having `this.pointList.depthBuffered = false` was somehow why the pointList points could be drawn without the transparent pixels being black.  (i don't know). and so maybe setting it to `true` is technically breaking the renderer. and so  `gl.depthMask(false)` partially fixes it (but apparently not always)

**Comment by Carifio24** [comment]
I just pushed a commit that should resolve the issue with the point representation of the Moon blocking the Artemis point. The idea was to basically copy the `depthMask` scheme from the spreadsheet layer into the point planet drawing as well.

This required an update to the patch script, so you'll need to clear things out and re-run it:
```bash
rm -rf node_modules; yarn; ./patch.sh
```
